### PR TITLE
[Radio] Design Guidelines

### PR DESCRIPTION
**Pull request by jesskuo4**
## Summary

Added `Radio` Design Guidelines

## List of notable changes:

- added new ComponentLayout to Radio docs
- added design guidelines

## What should reviewers focus on?

- 👀🙌  documentation: https://primer-b7c30e4a11-26139705.drafts.github.io/components/Radio

## Supporting resources (related issues, external links, etc):

- Epic: https://github.com/github/primer/issues/2433
- Radio component: https://github.com/primer/react/issues/1050
- [Brand] Checkbox: design guidelines: https://github.com/github/primer/issues/1996

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

 </td>
<td valign="top">

</td>
</tr>
</table>

### CODE REVIEW

File: apps/docs/content/components/Radio/index.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    <img src="https://github.com/primer/brand/assets/912236/7890cbc2-e588-4399-b903-78f2e581d05b" />
+    <Caption>
+        Don't overlap or leave out potential Radio options.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/912236/eb8c5797-1704-4a92-8f9f-69c0b895f2b7" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:42]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/766c9c24-6f02-444e-aad9-26dfabb8990d" />
```

File: apps/docs/content/components/Radio/index.mdx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    <Caption>
+        When used in groups, order Radios in logical order, like smallest to largest or least restrictive to most restrictive. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/912236/7890cbc2-e588-4399-b903-78f2e581d05b" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:48]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/9e125629-675c-4bc7-99eb-0e39a5a1de5d" />
```

File: apps/docs/content/components/Radio/index.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+description: Use the radio component when a user needs to select one option from a list.
+---
+
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+
+## Anatomy
+
+![An image displaying a Radio with a radio and label.](https://github.com/primer/brand/assets/912236/719022fa-4694-42c0-8554-b1923c65739f)
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:12]:
```suggestion
![An image displaying a Radio with a radio and label.](https://github.com/primer/brand/assets/131988618/1b494cdb-4991-4fc9-beac-feb02b296cb9)
```

File: apps/docs/content/components/Radio/index.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+The Radio component enables users to select a single option when used in a grop, making them suitable for scenarios where users are restricted to choosing just one option.
+
+For scenarios that allow for multiple selections, use the Checkbox(/components/Checkbox) component instead. To implement Radio buttons effectively, ensure that each radio button has a clear and concise label associated with it. Maintain consistent styling, including capitalization and without ending punctuation.
+
+If you have a short list of options and want users to select only one, consider using a radio button in a radio group. A radio group is a collection of related radio buttons grouped together. It helps provide clear context and organization for the options.
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/912236/eb8c5797-1704-4a92-8f9f-69c0b895f2b7" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:27]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/dff47404-e323-49d9-95f7-ab457a7195ef" />
```

File: apps/docs/content/components/Radio/index.mdx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    <Caption>
+        Ensure Radio options don't overlap and include "None" or "Other" options when necessary. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/912236/7890cbc2-e588-4399-b903-78f2e581d05b" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:33]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/38fb8666-f785-46ee-b3e3-9768ac8d0193" />
```

File: apps/docs/content/components/Radio/index.mdx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/dff47404-e323-49d9-95f7-ab457a7195ef" />
+    <Caption>
+        Ensure Radio options don't overlap and include "None" or "Other" options when necessary. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/38fb8666-f785-46ee-b3e3-9768ac8d0193" />
+    <Caption>
+        Don't overlap or leave out potential Radio options.
```
Comment (josepmartins) [inline apps/docs/content/components/Radio/index.mdx:35]:
```suggestion
        Don't overlap or leave out potential radio options from the list.
```

File: apps/docs/content/components/Radio/index.mdx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+- Radio: A selection control.
+- Label: A description for the radio. 
+
+## Usage
+
+Use the radio component to enable users to select a single option from a list.
+
+For scenarios that allow for multiple selections, use the [checkbox](/components/Checkbox) component instead. Keep the radio label descriptive and concise: aim for no more than 3 words. Write labels in sentence case and without ending in punctuation.
+
+### Grouping radio components
```
Comment (josepmartins) [inline apps/docs/content/components/Radio/index.mdx:23]: Suggestion to simplify as per page context 👍 

```suggestion
### Grouping
```

File: apps/docs/content/components/Radio/index.mdx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+Use the `name` prop to group together related radio components in a list.
+
+If you're not building something custom, you should use the `Stack` and `FormControl` component to render a group of radio inputs.
```
Comment (josepmartins) [inline apps/docs/content/components/Radio/index.mdx:26]: I think we could also simplify this by just saying that you could group related radio component in a list, using Stack and the FormControl components. I'd rather prefer to limit the prop exposure in the usage guidelines and leave it for the React section.

File: apps/docs/content/components/Radio/index.mdx [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/dff47404-e323-49d9-95f7-ab457a7195ef" />
```
Comment (josepmartins) [inline apps/docs/content/components/Radio/index.mdx:30]: Heads up this images uses an incorrect label style (it should be font weight bold)

File: apps/docs/content/components/Radio/index.mdx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+  src="https://user-images.githubusercontent.com/2313998/224133514-c583baf4-6dfc-444d-8714-dfc88cc00ff9.png"
+/>
+
+### Accessibility 
+
+Please use [FormControl](/components/FormControl) to render a standard radio input field. If using standalone, it should always be accompanied by a corresponding `<label>` to improve support for assistive technologies.
+
+## Options
+
+### Caption
```
[the remark this answers is not in the document]
Comment (josepmartins) [reply apps/docs/content/components/Radio/index.mdx:70]: I think you miss to remove this part right?

File: apps/docs/content/components/Radio/index.mdx [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224133514-c583baf4-6dfc-444d-8714-dfc88cc00ff9.png"
+/>
+
+## Options
+
+### Caption
+
+A caption for a radio can be useful in certain scenarios where additional information or clarification is needed for the radio option. Some examples include complex or technical options, ambiguous or unclear options, legal or compliance requirements, or accessibility considerations.
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:72]:
```suggestion
```

File: apps/docs/content/components/Radio/index.mdx [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+description: Use the radio component when a user needs to select one option from a list.
+---
+
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+
+## Anatomy
+
+![An image displaying a Radio with a radio and label.](https://github.com/primer/brand/assets/131988618/1b494cdb-4991-4fc9-beac-feb02b296cb9)
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:12]:
```suggestion
![An image displaying a Radio with a radio and label.](https://github.com/primer/brand/assets/131988618/d25dafb9-87f8-4b6a-a205-57a851d47b5e)
```

File: apps/docs/content/components/Radio/index.mdx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+Use the radio component to enable users to select a single option from a list.
+
+For scenarios that allow for multiple selections, use the [Checkbox](/components/Checkbox) component instead. Keep the radio label descriptive and concise: aim for no more than 3 words. Write labels in sentence case and without ending in punctuation.
+
+### Grouping
+To group related radio components together in a list, use the [Stack](/components/Stack) and [FormControl](/components/FormControl) components. When rendering a group of radio components, make sure to consider all potential option. Remember to include "None" or "Other" options when necessary. 
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/dff47404-e323-49d9-95f7-ab457a7195ef" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:28]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/151c50cb-c3ac-4790-901d-a5be38fad46e" />
```

File: apps/docs/content/components/Radio/index.mdx [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+    <Caption>
+        Ensure radio options are are mutually exclusive. Include "None" or "Other" options when necessary. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/38fb8666-f785-46ee-b3e3-9768ac8d0193" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:34]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/5d25cfee-bf8e-4b93-93da-5dd40f38f3be" />

```

File: apps/docs/content/components/Radio/index.mdx [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+    <Caption>
+        Don't overlap or leave out potential radio options from the list.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/766c9c24-6f02-444e-aad9-26dfabb8990d" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:43]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/433d5deb-0395-4d24-8f5d-513178d9e944" />


```

File: apps/docs/content/components/Radio/index.mdx [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+    <Caption>
+        When used in groups, order the radio options in a logical order. E.g. smallest to largest or least restrictive to most restrictive options. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/9e125629-675c-4bc7-99eb-0e39a5a1de5d" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:49]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/1f4c4e50-ef1a-44f1-876c-54ca2036fddf" />

```

File: apps/docs/content/components/Radio/index.mdx [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+
+### States
+
+Radios have static dimensions and three different states: default, checked, and disabled.
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224133514-c583baf4-6dfc-444d-8714-dfc88cc00ff9.png"
+/>
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:64]:
```suggestion
<img src="https://github.com/primer/brand/assets/131988618/94a893a6-0063-4204-a7ab-da7b5aaaa96b" />

```

File: apps/docs/content/components/Radio/index.mdx [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/433d5deb-0395-4d24-8f5d-513178d9e944" />
+    <Caption>
+        When used in groups, order the radio options in a logical order. E.g. smallest to largest or least restrictive to most restrictive options. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/1f4c4e50-ef1a-44f1-876c-54ca2036fddf" />
+    <Caption>
+        Don't order radio options without logic in a list.
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:51]:
```suggestion
        Don't order radio options without logic in a group.
```

File: apps/docs/content/components/Radio/index.mdx [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+        Don't overlap or leave out potential radio options from the list.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/433d5deb-0395-4d24-8f5d-513178d9e944" />
+    <Caption>
+        When used in groups, order the radio options in a logical order. E.g. smallest to largest or least restrictive to most restrictive options.
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:45]:
```suggestion
        When used in groups, order the radio options in a logical order (e.g. smallest to largest).
```

File: apps/docs/content/components/Radio/index.mdx [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/433d5deb-0395-4d24-8f5d-513178d9e944" />
+    <Caption>
+        When used in groups, order the radio options in a logical order. E.g. smallest to largest or least restrictive to most restrictive options. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/1f4c4e50-ef1a-44f1-876c-54ca2036fddf" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Radio/index.mdx:49]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/9d3f4639-c064-4b18-bcce-89a05a29b4e1" />

```

**Review by josepmartins — APPROVED** [review]
Looking good @jesskuo4!! 👍 good job

**Pull request merged**